File: src/day1.py

def parse_calories(string):
    elves = [[int(str) for str in elf.split("\n")]
             for elf in string.strip().split("\n\n")]
    return [sum(elf) for elf in elves]


def most_calories(string):
    return max(parse_calories(string))


def three_most_calories(string):
    totals = reversed(sorted(parse_calories(string)))
    return sum(list(totals)[0:3])


with open("./inputs/day1.txt") as f:
    content = f.read()
    print(most_calories(content))
    print(three_most_calories(content))

File: src/day1.ts
import * as fs from "fs";

const sum: (list: number[]) => number = (list) => list.reduce((acc, elem) => acc + elem);
const readInput: (day: number) => string = (number) => fs.readFileSync(`./inputs/day${number}.txt`).toString();

function stringToElfCaloriesTotal(str: string): number[] {
  return str
    .trim()
    .split("\n\n")
    .map((str) => str.split("\n"))
    .map((list) => list.map((x) => parseInt(x)))
    .map((elem) => sum(elem));
}

function mostCalories(str: string): number {
  const calories = stringToElfCaloriesTotal(str);
  return Math.max(...calories);
}

function threeMostCalories(str: string): number[] {
  return stringToElfCaloriesTotal(str)
    .sort((a, b) => b - a)
    .slice(0, 3);
}

const input = readInput(1);

console.log(mostCalories(input.toString()));
console.log(sum(threeMostCalories(input.toString())));
